ncard_app/views/filters.py

```diff
@@ -45,19 +45,22 @@ def universal_search(self, queryset, name, value):
 
 class OrganisationFilter(django_filters.FilterSet):
     query = django_filters.CharFilter(method='universal_search',
-                                      label="name")
+                                      label="name, type")
 
     class Meta:
         model = models.Organisation
         fields = {}
 
     def universal_search(self, queryset, name, value):
-        return models.Organisation.objects.all().filter(
-
-            Q(name__icontains=value)
+        qs = Q(name__icontains=value)
 
-        )
+        country_reverse = dict((v, k) for k, v in models.Organisation.OrganisationType.choices)
+        for key in country_reverse.keys():
+            if value.lower() in key.lower():
+                dict_value = country_reverse[key]
+                qs.add(Q(organisation_type=dict_value), Q.OR)
```

Comment:
The dict shouldn't be called country_reverse when it has nothing to do with countries.

In addition, why bother reversing the dictionary when you're just going to iterate over the whole thing anyway?

```suggestion
        for index, choice_name in models.Organisation.OrganisationType.choices:
            if value.lower() in choice_name.lower():
                qs.add(Q(organisation_type=index), Q.OR)
```